docs/contributing/code_guide.rst

```diff
@@ -279,39 +279,138 @@ to edit the pull request by changing the corresponding branch on your
 PlasmaPy fork on GitHub.  After a pull request is merged into the
 code, you may delete the branch you created for that pull request.
 
-Commit Messages
----------------
-Good commit messages communicate context and intention to other
-developers and to our future selves.  They provide insight into why we
-chose a particular implementation, and help us avoid past mistakes.
 
-Suggestions on `how to write a git commit message
-<https://cbea.ms/git-commit>`_:
+Comments
+========
+
+A well-placed and well-written comment can prevent future frustrations.
+However, comments are not inherently good. As code evolves, an
+unmaintained comment may become outdated, or get separated from the
+section of code that it was meant to describe. Cryptic and obsolete
+comments may end up confusing contributors. In the worst case, an
+unmaintained comment may contain inaccurate or misleading information
+(hence the saying that "a comment is a lie waiting to happen").
+
+* Refactor code to make it more readable, rather than explaining how it
+  works :cite:p:`wilson:2014`.
+
+* Instead of using a comment to define a variable, rename the variable
+  to encode its meaning and intent.  For example, code like:
+
+  .. code-block:: python
+
+     # collision frequency
+     nu = 1e6 * u.s ** -1
+
+  could be achieved with no comment by doing:
+
+  .. code-block:: python
+
+     collision_frequency = 1e6 * u.s ** -1
+
+* Use comments to communicate information that you wish you knew before
+  starting to work on a particular section of code, including
+  information that took some time to learn.
+
+* Use comments to communicate information that the code cannot,
+  such as why an alternative approach was *not* taken.
+
+* When possible, include enough contextual information that the comment
+  will make sense even if it is displaced from the code it was
+  originally describing.
+
+* Use comments to include references to books or articles that describe
+  the equation, algorithm, or software design pattern that is being
+  implemented. Even better, include these references in docstrings.
+
+* Provide enough contextual information in the comment for a new user
+  to be able to understand it.
+
+* Remove commented out code before merging a pull request.
+
+* When updating code, be sure to update the comments too!
+
+* When a comment is used as the header for a section of code, consider
```

Comment:
You're going to make me rage quit with this point. 😂 I'd like to see some kind of comment in here that there's a "subjective" balancing act with the `extract function refactoring pattern`.  Overall it's good, but when used on a too fine of a scale it can significantly reduce code readability.

Reply:
That's a good point!  This is a good place to put in a tradeoff analysis and how there are different factors we need to balance.

Reply:
Added in a discussion on this below.